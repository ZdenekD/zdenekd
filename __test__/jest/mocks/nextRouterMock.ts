const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        asPath: '/page/subpage/something-else',
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
