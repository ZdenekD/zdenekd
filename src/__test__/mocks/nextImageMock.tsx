const nextImageMock = jest.mock('next/image', () => function Image() {
    return <img src="" alt="" />;
});

export default nextImageMock;
