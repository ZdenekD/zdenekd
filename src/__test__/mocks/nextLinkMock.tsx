const nextLinkMock = jest.mock('next/link', () => function Link() {
    return <a href="/">link</a>;
});

export default nextLinkMock;
