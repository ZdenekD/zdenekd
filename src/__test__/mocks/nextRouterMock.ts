const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
