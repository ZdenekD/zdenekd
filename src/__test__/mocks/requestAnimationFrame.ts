import react from 'react';

window.requestAnimationFrame = () => 0;

export default react;
