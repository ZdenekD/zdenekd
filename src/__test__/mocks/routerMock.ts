const routerMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
    })),
}));

export default routerMock;
