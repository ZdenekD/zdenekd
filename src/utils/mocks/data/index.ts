import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
