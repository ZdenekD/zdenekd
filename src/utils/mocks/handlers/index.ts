import handler from './handler';

const handlers = [...handler];

export default handlers;
