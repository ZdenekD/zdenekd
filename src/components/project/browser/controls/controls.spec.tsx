import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../../../__test__/utils/helpers';
import '../../../../__test__/mocks/nextRouterMock';
import Controls from '.';

describe('Controls', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Controls
                isFirst={false}
                isLast={false}
            />
        );
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const prev = findComponent(wrapper, 'component-controls-prev');
        const next = findComponent(wrapper, 'component-controls-next');

        expect(prev.exists()).toBe(true);
        expect(next.exists()).toBe(true);
    });

    it('triggers handlePrev function on prev button click', () => {
        const mockFunction = jest.fn();
        const container = shallow(
            <Controls
                isFirst={false}
                isLast={false}
                handlePrev={mockFunction}
            />
        );
        const component = findComponent(container, 'component-controls-prev');

        component.simulate('click');

        expect(mockFunction).toHaveBeenCalled();
    });

    it('triggers handleNext function on next button click', () => {
        const mockFunction = jest.fn();
        const container = shallow(
            <Controls
                isFirst={false}
                isLast={false}
                handleNext={mockFunction}
            />
        );
        const component = findComponent(container, 'component-controls-next');

        component.simulate('click');

        expect(mockFunction).toHaveBeenCalled();
    });
});
