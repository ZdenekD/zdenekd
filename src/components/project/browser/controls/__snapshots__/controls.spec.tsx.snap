// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls match snapshot 1`] = `
<button type="button"
        class="prev "
        aria-label="Previous project"
        tabindex="0"
        data-test="component-controls-prev"
>
</button>
<button type="button"
        class="next "
        aria-label="Next project"
        tabindex="0"
        data-test="component-controls-next"
>
</button>
`;
